src/pages/index.js

```diff
@@ -135,30 +224,107 @@ function closeOverlay(event) {
   }
 }
 
-modals.forEach((modal) => {
-  modal.addEventListener("mousedown", closeOverlay);
-});
+// modals.forEach((modal) => {
+//   modal.addEventListener("mousedown", closeOverlay);
+// });
 
 function handleEditFormSubmit(event) {
   event.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
 
-  closeModal(editModal);
+  //change text content to "Saving..."
+  const submitButton = event.submitter;
+  submitButton.textContent = "Saving...";
+  setButtonText(submitButton, true);
+  // setButtonText(cardSubmitButton, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      // use data argument instead of the input values
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // call setButtonText instead
+      setButtonText(submitButton, false);
+    });
 }
 
+// implement loading text for all other form submissions
+
 function handleAddCardSubmit(event) {
   event.preventDefault();
 
   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
+  const cardModalSubmitButton = event.submitter;
+  setButtonText(cardModalSubmitButton, true, "Save", "Saving...");
+
+  // cardsList.prepend(cardElement);
+  // event.target.reset();
+
+  api
+    .postCard(inputValues)
+    .then((card) => {
+      const cardElement = getCardElement(inputValues);
+      cardsList.prepend(cardElement);
```

Comment:
* Likes don't work https://snipboard.io/g9aUJH.jpg
* I can't delete **new** cards https://snipboard.io/8LM4x1.jpg 


Please, use the `card` from line 272 because it contains `_id` which is needed for liking and deleting


```js
    .then((card) => {
      const cardElement = getCardElement(card);
      cardsList.prepend(cardElement);

```